integrations/weaviate/tests/test_document_store.py

```diff
@@ -625,6 +625,121 @@ def test_embedding_retrieval_with_distance_and_certainty(self, document_store):
+                Document(content="C# is a functional programming language", embedding=[0.6, 0.4, 0.6, 0.5]),
+            ]
+        )
+        result = document_store._hybrid_retrieval("functional", query_embedding=[1.0, 0.8, 0.2, 0.1], top_k=3)
+        assert len(result) == 3
+        assert all("functional" in doc.content for doc in result)
+        assert all(doc.score is not None and doc.score > 0.0 for doc in result)
+
+    def test_hybrid_retrieval_with_alpha(self, document_store):
+        document_store.write_documents(
+            [
+                Document(content="Haskell is a functional programming language", embedding=[1.0, 0.8, 0.2, 0.1]),
+                Document(content="Lisp is a functional programming language", embedding=[0.9, 0.7, 0.3, 0.2]),
+                Document(content="C++ is an object oriented programming language", embedding=[0.1, 0.2, 0.8, 0.9]),
+            ]
+        )
+        # Test with alpha=0.0 (pure BM25)
+        result_bm25 = document_store._hybrid_retrieval("functional", query_embedding=[1.0, 0.8, 0.2, 0.1], alpha=0.0)
+        assert len(result_bm25) > 0
+        assert result_bm25[0].score > 0.0
+
+        # Test with alpha=1.0 (pure vector search)
+        result_vector = document_store._hybrid_retrieval("functional", query_embedding=[1.0, 0.8, 0.2, 0.1], alpha=1.0)
+        assert len(result_vector) > 0
+        assert result_vector[0].score > 0.0
+
+        # Test with alpha=0.5 (balanced hybrid)
+        result_hybrid = document_store._hybrid_retrieval("functional", query_embedding=[1.0, 0.8, 0.2, 0.1], alpha=0.5)
+        assert len(result_hybrid) > 0
+        assert result_hybrid[0].score > 0.0
+
+    def test_hybrid_retrieval_with_max_vector_distance(self, document_store):
+        document_store.write_documents(
+            [
+                Document(content="Haskell is a functional programming language", embedding=[1.0, 0.8, 0.2, 0.1]),
+                Document(content="Lisp is a functional programming language", embedding=[0.9, 0.7, 0.3, 0.2]),
+                Document(content="C++ is an object oriented programming language", embedding=[0.1, 0.2, 0.8, 0.9]),
+            ]
+        )
+        # Use a restrictive max_vector_distance to limit results
+        result = document_store._hybrid_retrieval(
+            "functional", query_embedding=[1.0, 0.8, 0.2, 0.1], max_vector_distance=0.5
+        )
+        assert len(result) >= 1  # Should find at least the closest match
+        assert all(doc.score is not None and doc.score > 0.0 for doc in result)
+
+    def test_hybrid_retrieval_empty_query(self, document_store):
+        document_store.write_documents(
+            [
+                Document(content="Test document", embedding=[1.0, 0.8, 0.2, 0.1]),
+            ]
+        )
+        # Test with empty query string
+        result = document_store._hybrid_retrieval("", query_embedding=[1.0, 0.8, 0.2, 0.1])
+        assert len(result) >= 0  # Should handle empty query gracefully
```

Comment:
Would it be possible to bump this to `assert len(result) == 1` or would we expect it to be `0` if the query is empty?

Reply:
When alpha==0 it returns empty list. When alpha>0 it returns a non empty list.
Hence len(result) >= 0 is correct here just to check if it is not throwing errors.